--- cutter-1.04.orig/cutter.c
+++ cutter-1.04/cutter.c
@@ -613,27 +613,20 @@
 			(match(ip1,port1,src1n,sport1) && match(ip2,port2,dst1n,dport1)) ||
 			(match(ip1,port1,dst1n,dport1) && match(ip2,port2,src1n,sport1))
 		) {
-			/*
-			 * local network to public network - forwarded connection
-			 */
+			found ++;
 
 			if (!localip(src1n) && !localip(dst1n) && !localip(src2n) && localip(dst2n)) {
-				found ++;
-				add_to_connection_list(dst1, dport1, src1, sport1);
-				add_to_connection_list(dst2, dport2, src2, sport2);
+				/* local network to public network - nated connection */
 			}
-
-			/* Inbound connection forwarded to private network device */
-
 			else if (!localip(src1n) && localip(dst1n) && !localip(src2n) && !localip(dst2n)) {
-				found ++;
-				add_to_connection_list(dst1, dport1, src1, sport1);
-				add_to_connection_list(dst2, dport2, src2, sport2);
+				/* inbound connection nated to private network device */
+			}
+			else {
+				/* non nat'ed connection */
 			}
 
-			/* We appear to be running in the server or client, not in a router/firewall */
-			else
-				found_but_local ++;
+			add_to_connection_list(dst1, dport1, src1, sport1);
+			add_to_connection_list(dst2, dport2, src2, sport2);
 		}
 	}
 
